### ISSUE TITLE
Analytics Implementation Search Type-Ahead POC

### ISSUE THREAD

**Issue by johnhashva** [department-of-veterans-affairs/va.gov-team#13494]
# Google Analytics Implementation

## Description
Search.gov has recently introduced "type-ahead" via its API, enabling VA.gov to bring this long-awaited "table stakes" capability to our users.  

Our upfront audit has revealed that the type-ahead suggestions powered by Search.gov are not as "relevant/valuable" as expected. Further investigation by Search.gov has determined that the feature logic is not working as expected and is under active investigation for resolution.  

**Despite this, we believe feature "as is" can be deployed as a Proof-of-Concept to a small share of VA.gov users for a period of up to four weeks to glean insights about the effectiveness of this new utility.**.  More [details are here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8896).

Our hypothesis:
 - We believe that enabling type-ahead as a search feature
 - Will result In more relevant search results for users
 - We will know we have succeeded when we see a significant decrease in bounce rate compared to the overall site search average of 48% and a decrease in paging associated with search.

## POC Considerations

**_To be tracked (possibilities):_**
- What percentage of users search by using the type-ahead feature?
- What terms are users querying when they use type-ahead?
- What suggestions are users clicking on?
- What is the percentage of suggestions clicked vs. ignored?
- What is the bounce rate from pages clicked to from the type-ahead results/suggestions?

_**Questions to answer:**_
- Does type-ahead reduce "wayward queries" by offering a shortcut to the most popular and/or trending topics/content (e.g. instead of a general phrase such as "disability" type-ahead suggests "disability rating," "disability ratings," etc.)?
- Does type-ahead reduce search bounce rate?

## Stakeholders

- Who is this request for?
    - John Hashimoto @johnhashva 
    
- What team/project is this work for?
    - Public Websites - [Search Initiative 2020-Onsite](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/public-websites/onsite-search)
    
- Provide the Domain where Google Analytics needs to be implemented
    - www.va.gov (type-ahead is currently deployed on the HP of staging.va.gov)

- Please provide any upcoming deadlines or important milestones
    - We would like to launch this POC to 10% of VA.gov users (utilizing Flipper) in September.

> Please leave the following blank

## Acceptance Criteria
- [ ] QA has been completed
- [ ] VFS team completes brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed

**Comment by jonwehausen** [comment]
Thank you @johnhashva ! Confirming receipt here

**Comment by johnhashva** [comment]
@jonwehausen and @bsmartin-ep and @bmcgrady-ep  here is the VA.gov search "audit" I conducted on both Type-Ahead suggestions and search page results.   This should be helpful to you.

[Search.gov_Type-ahead_Issues.docx](https://github.com/department-of-veterans-affairs/va.gov-team/files/5225720/Search.gov_Type-ahead_Issues.docx)

**Comment by jonwehausen** [comment]
## Datalayer Specification 
cc: @johnhashva 

During the initial roll-out period, In order to make sure we understand those users receiving the type ahead functionality and those users still on the old experience...we would like to request an event be pushed on page load: 
```javascript
   'event': 'search-type-ahead-enabled' //or 'search-type-ahead-disabled'
```

From there, we would like to propose the following specification on **each search:** 
```javascript
'event': 'onsite-search-performed'
'search-text-input': //populate with what the user typed in at the time they made their type ahead selection
'type-ahead-option-keyword-selected':  //populate with the option that was selected
'type-ahead-option-position':  //populate with the numerical position of the suggestion at the time selected 1-5
'type-ahead-options-list':  //populate with array of all five suggestions that were available
```

### Example
Description and Screenshot  |  Data Layer Sample
------------ | -------------
Typed `'heal'` and selected `'health assessment'` and searched ![image](https://user-images.githubusercontent.com/48527022/93533545-2b092000-f911-11ea-82bc-8a3a45ce806b.png) | `'event': 'onsite-search-performed'`<br>`'search-text-input': 'heal',`<br>`'type-ahead-option-keyword-selected': 'health assessment',`<br>`'type-ahead-option-position': 2,`<br>`'type-ahead-options-list': {'healing touch', 'health assessment', 'health problems that i have', 'health coverage statement 2019', 'health topics'}`


### Developer Notes
- The state of these variables will need to be stored prior to the search button being clicked, but this event should **ONLY** fire when the search has actually been performed to conserve hit volume to actual searches
- These variables should reset each time a new option in the list is selected
- For `type-ahead-option-keyword-selected`, populate with **`undefined`** if the user hasn't clicked on any option and performed the search organically
- ⚠️ **EDGE CASE**: A user types in their search, selects an option, types some more into their query, and searches where the final search query doesn't match the suggestion... @johnhashva I will let you make the call if this type of scenario that should be recorded as having used the suggestions or count this as an organic search -- happy to discuss this edge case further

**Comment by johnhashva** [comment]
@jonwehausen going to schedule 30m for us (including @ncksllvn and @MickinSahni) to get a walk-through of this approach and pose any questions.  Will look at next week.

**Comment by jonwehausen** [comment]
Sure thing!

**Comment by johnhashva** [comment]
Follow-ups from  9/29 meeting with @jonwehausen:

- For Edge Case outlined above, let's treat that as an organic search event.
- @ncksllvn can instrument as is -- no need for the option of a second tracking event.  That said, @jonwehausen will check to see how easy this would be to create.
- @jonwehausen to confer with brian about feasibility of deeper funnel analysis -- e.g. the user invokes type-ahead, picks a suggestion, lands on Search Results page for that suggested keyword/phrase, clicks on a results page link OR  refines their search OR bounces from page.

**Comment by jonwehausen** [comment]
Hi team, 

Thanks again for your time yesterday @johnhashva , @ncksllvn and @MickinSahni!

**After discussing internally, we would like to add an additional event to track search result clicks, please find the sample event below:** 

```javascript
'event': 'onsite-search-results-click',
'search-query': //query that is currently in the search box at time of clicking on result
'search-total-results': //populate with numerical total number of results
'search-total-result-pages': //populate with total number of result pages returned
'search-result-position':  //populate with the numerical position of the search result, relative to the current result page
'search-result-page':  //populate with the current numerical page number the user clicked the result from (pagination number) 
```


### Example
Description and Screenshot  |  Data Layer Sample
------------ | -------------
Reached search page ![image](https://user-images.githubusercontent.com/48527022/94718728-246ea580-0320-11eb-932f-56b480bd5040.png)  | `'event': 'onsite-search-results-click',`<br> `'search-query': 'health assessment',`<br>`'search-total-results': 999,`<br>`'search-total-result-pages': <total_result_pages>,`<br>`'search-result-position':  4`<br>`'search-result-page': 1`

@johnhashva Please let us know any further questions you have around this tracking.

**Comment by johnhashva** [comment]
Thanks @jonwehausen   We will take a closer look when @ncksllvn has some time and get back with any q's.

**Comment by jonwehausen** [comment]
@bsmartin-ep Next steps on our part are to stage the GTM configuration according to this specification

@ncksllvn @johnhashva @MickinSahni keep us posted on the status of the dataLayer implementation here-- looking forward to these analytics.

**Comment by denisecoveyduc** [comment]
To be assigned to Sam; to work with Nick Sullivan. Assigning for Sprint 34

**Comment by denisecoveyduc** [comment]
@srsuddath  is reviewing this ticket and believes it will require some rework. It is considered a stretch goal for sprint 34.

**Comment by ncksllvn** [comment]
To add - I don't know if we documented it or not, but I recall a convo between me, @johnhashva, @jonwehausen, and @bsmartin-ep where we decided that the current typeahead behavior should be changed so that after clicking a suggestion, the input is submitted automatically. This way, the user doesn't have to press a suggestion, observe it in the input box, then press enter/click submit - all they have to do is click a suggestion. This is more inline with a typical typeahead and reduces the edge cases around analytics - "what if they type, click a suggestion, then type some more before submitting?"

**Comment by ncksllvn** [comment]
@johnhashva can you confirm the above? If that is a change we need to make, should it be done as part of this ticket (additional acceptance criteria) or in a separate one? Apologies if there already is one and I'm just creating needless confusion 🙂

**Comment by johnhashva** [comment]
Confirmed @ncksllvn  -- and I should have ticketed this.  @srsuddath and I just chatted about this and we will make sure to include this as part of our POC release.  Thanks!

**Comment by jonwehausen** [comment]
Thanks for resurfacing that conversation and feature of the typeahead functionality @ncksllvn! You're certainly correct that change would help eliminate edge cases in collection & interpretation of the search data as well. 

Just an fyi here as well @denisecoveyduc @srsuddath @ncksllvn & @johnhashva -- we made small a tweak to the [specification above ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13494#issuecomment-701534639) to change the event name of **`'on-site-search-performed'`** to **`'view_search_results'`** to comply and prepare for upcoming GA4 taxonomy changes. 

When type ahead is used, we will set the `'type-ahead-option-keyword-selected'` to populate with that value selected and when it's not used and search is done organically we set `undefined` for that value. 

Let us know if you all would like to have a follow-up conversation around this specification. Thanks again for your work here.

**Comment by denisecoveyduc** [comment]
11/18- Changed sizing to align with new estimates.

**Comment by johnhashva** [comment]
@jonwehausen (with @denisecoveyduc + @srsuddath + @ncksllvn FYSA) -- we've decided to launch the type-ahead POC site wide (vs. just home page) to ~5% of audience for 4 weeks.  With that change in mind: Will we be able to track where Type Ahead is used since we are making it global -- or is this a new dimension to the spec? (Apologies for the small creep here).

**Comment by jonwehausen** [comment]
Hi team, 

Thanks for the notice & update here!

**A couple items here:** 
- To understand where the user performed the search, I would recommend we pass in a `'search-page-path'` variable indicating where the search took placed along with each `'view_search_results'` event. You can find that specification [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13494#issuecomment-694532923). 
- In addition, in order to understand the 5% of users who are receiving the type ahead search functionality, I would recommend we push an event to the datalayer once the system knows the user has type ahead enabled. This will then allow us to segment these users to improve accuracy of usability & success metrics. 
```javascript
'event': 'phased-roll-out-enabled', 
'type-ahead-enabled': true|false
```

Let us know any further questions you have here! Thanks 
cc: @johnhashva @srsuddath @denisecoveyduc @ncksllvn

**Comment by denisecoveyduc** [comment]
As per slack discussion on 11/18, this will become a stretch goal because of the addition of #14955 to the sprint. @johnhashva @srsuddath

**Comment by srsuddath** [comment]
@denisecoveyduc almost done with the "face-lift" of the typeahead functionality while I wait on Search.gov to get me the API key. Depending on the timeline for that, I will try to get the google analytics tagging working tomorrow. Then all that will be remaining here is styling and accessibility!

**Comment by jonwehausen** [comment]
Thanks for the update @srsuddath , please keep us posted on when this will be ready in staging for our testing, thanks again for your work here!

**Comment by srsuddath** [comment]
@denisecoveyduc, @jonwehausen I'd like to propose pulling the following comment off and splitting the required work into its own ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/13494#issuecomment-701534639

Where this ticket focuses on the Typeahead POC i'd like to have a separate issue to track changes made to the search results page.

**Comment by johnhashva** [comment]
@jonwehausen Howdy, @srsuddath is working through (almost done) on the instrumentation as outlined here -- but we have a few questions re: the Search Results dimension -- e.g,  add an additional event to track search result clicks (apologize if we covered this, memory is hazy): With this event tagging, **will we be able to glean/understand the following flow (the type ahead conversion funnel, so to speak)** -- 

- Percentage of users who take action with Type Ahead and the percentage who ignore Type Ahead (e.g. they typed a term, but ignored the suggestions)
- For those who took action with Type Ahead (clicked on one of the suggestions), how many:
  + What term the user queried 
  + What suggestions the user saw, and which suggested term they clicked
  + What the user did on the resulting search results page
     - They clicked on one of the displayed results, or ...
     - They refined their search
     - They bounced from the search results page (did not click any of the results)

LMK if this level of granularity is not possible -- and what aspects of this flow we will be able to track.    (Happy to Zoom on this topic with @srsuddath and @denisecoveyduc if better).

**Comment by jonwehausen** [comment]
Hi @johnhashva , 

Great questions here! To answer each respectively: 
> Percentage of users who take action with Type Ahead and the percentage who ignore Type Ahead (e.g. they typed a term, but ignored the suggestions)
- Yes, we will be able to formulate these percentages based on the volume of `'view_search_results'` events where the data layer variable `'type-ahead-option-keyword-selected'` is populated vs. not

> For those who took action with Type Ahead (clicked on one of the suggestions), how many:
> What term the user queried
- Yes
> What suggestions the user saw, and which suggested term they clicked
- Yes, this is where the data layer variables for `'type-ahead-options-list'` and `'type-ahead-option-keyboard-selected'` will come into play
> What the user did on the resulting search results page
- Yes, the event `'onsite-search-results-click'` will be utilized to understand search result page visits, (comment [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13494#issuecomment-701534639)) along with other metadata such as `'search-result-position'` and `'search-result-page'` to give greater context to search usability and performance metrics. 
- @srsuddath I believe this was the spec you want to implement in a separate iteration per your [latest note here. ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13494#issuecomment-734942717)

> They clicked on one of the displayed results, or ...
- Yes
> They refined their search
- Yes, this is an area I would recommend we provide thru BigQuery for greater precision. 
> They bounced from the search results page (did not click any of the results)
- Yes

Hope this helps and let us know any further questions you may have.

**Comment by srsuddath** [comment]
current list of events I am logging:
On Page Load:
```
event: 'phased-roll-out-enabled',
      'type-ahead-enabled': this.props.searchTypeaheadEnabled,
```

On Typeahead Search 
```
event: 'view_search_results',
      'search-page-path': which page did you search from,
      'search-text-input': user input,
      'type-ahead-option-keyword-selected': chosen suggestion,
      'type-ahead-option-position': array position,
      'type-ahead-options-list': list of suggestions,
```
On Non-Typeahead Search
```
event: 'view_search_results',
      'search-page-path': which page did you search from,
      'search-text-input': user input,
      'type-ahead-option-keyword-selected': undefined,
      'type-ahead-option-position': undefined,
      'type-ahead-options-list': list of suggestions (if there are any),
```

**Comment by srsuddath** [comment]
And on the following page when a user clicks a search result, this is what we are tagging:
```
event: 'onsite-search-results-click',
      'search-query': user input,
      'search-total-results': number of results,
      'search-total-result-pages': number of pages,
      'search-result-position': where on the page the result they clicked is,
      'search-result-page': which page of search results they are on,
```

**Comment by srsuddath** [comment]
@jonwehausen I'd like to propose a change to the events we are logging (note the addition of the flag at the end)

```
event: 'view_search_results',
      'search-page-path': which page did you search from,
      'search-text-input': user input,
      'type-ahead-option-keyword-selected': undefined,
      'type-ahead-option-position': undefined,
      'type-ahead-options-list': list of suggestions (if there are any),
      'search-typeahead-enabled': true | false,
```

**Comment by jonwehausen** [comment]
Great suggestion @srsuddath! 

Let us know when we will be able to QA in staging.

**Comment by jonwehausen** [comment]
Expecting code to be moved to staging today, goal is to have launched on Friday. 

GTM Workspace: https://tagmanager.google.com/#/container/accounts/144676348/containers/2214120/workspaces/1000366

**Placeholders were put in for custom dimension / metric index number**

@bsmartin-ep / @bmcgrady-ep do one of you have capacity for QA? Thanks

**Comment by bsmartin-ep** [comment]
### QA

Interaction|Data Layer|GTM Tag|Pass
-----|-----|-----|-----
`phased-roll-out-enabled` flag for user|N/A|N/A|N/A
View Search Results|![image](https://user-images.githubusercontent.com/62304138/101299783-65dc6f00-3801-11eb-856f-1b740cae9593.png)|![image](https://user-images.githubusercontent.com/62304138/101299758-47767380-3801-11eb-9c88-30e234280c64.png)|**:heavy_check_mark: PASS**
Onsite Search Results Click|![image](https://user-images.githubusercontent.com/62304138/101299862-a89e4700-3801-11eb-89b6-d3a3c3898b63.png)|![image](https://user-images.githubusercontent.com/62304138/101299895-c66bac00-3801-11eb-9fb8-1f379ddb9a98.png)|**:heavy_check_mark: PASS**

------
Should I see the `phased-roll-out-enabled`event on staging @srsuddath?

**Comment by srsuddath** [comment]
@bsmartin-ep Yes, you should. Let me check around and see what I can find out.

**Comment by srsuddath** [comment]
@bsmartin-ep I found a bug that was preventing the error from being logged, I have pushed up a fix to this pull request:
https://github.com/department-of-veterans-affairs/vets-website/pull/15181

**Comment by bsmartin-ep** [comment]
@srsuddath -

`phased-roll-out-enabled` looks good.

 `search-text-input` has been changed to `search-query` in event `view_search_results`?

Thanks.

**Comment by bsmartin-ep** [comment]
One thing I noticed missing from Jon's workspace is the `search-selection` parameter.  Did you and Jon happen to discuss if that would be included in this implementation, @srsuddath?

This has been published to staging, so we [should see data in non-prod GA](https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w178298118p176884211/_u.date00=20201209&_u.date01=20201210&explorer-segmentExplorer.segmentId=analytics.eventLabel&explorer-table.plotKeys=%5B%5D&explorer-table.filter=view_search_results&explorer-table.secSegmentId=analytics.customDimension115&explorer-table.advFilter=%5B%5B0,%22analytics.eventLabel%22,%22RE%22,%22view_search_results%22,0%5D%5D).

### Re-QA using [Jon's workspace](https://tagmanager.google.com/#/container/accounts/144676348/containers/2214120/workspaces/1000366)

Interaction|Data Layer|GTM Tag|Pass
-----|-----|-----|-----
`phased-roll-out-enabled` flag for user|![image](https://user-images.githubusercontent.com/62304138/101576709-59921680-39a7-11eb-9b65-f31e25297adc.png)|![image](https://user-images.githubusercontent.com/62304138/101577517-93fbb380-39a7-11eb-86b5-598b6f252c77.png)|**:heavy_check_mark: PASS**
Site Search (no type ahead)|![image](https://user-images.githubusercontent.com/62304138/101842918-1a82d300-3b17-11eb-9bb0-72524d14126b.png)|![image](https://user-images.githubusercontent.com/62304138/101842915-15258880-3b17-11eb-9f22-6f400e75a8c8.png)|**:heavy_check_mark: PASS**
Site Search (with type ahead)|![image](https://user-images.githubusercontent.com/62304138/101842955-2d95a300-3b17-11eb-9035-cc09d542f9de.png)|![image](https://user-images.githubusercontent.com/62304138/101842971-3b4b2880-3b17-11eb-8f06-fac760a5516f.png)|**:heavy_check_mark: PASS**
Onsite Search Results Click|![image](https://user-images.githubusercontent.com/62304138/101843169-a09f1980-3b17-11eb-9bf3-d36a10f2cb2f.png)|![image](https://user-images.githubusercontent.com/62304138/101843205-b14f8f80-3b17-11eb-999f-e7f26a2173ce.png)| ??????

---------------


New Custom Dimensions added:
-----

114     Type Ahead Option Keyword Selected<br>
115     Type Ahead Options List<br>



New Custom Metrics added:
------  
39     Type Ahead Option Position<br>
40     Search Results Psition<br>
41     Search Results Total Count<br>
42     Search Results Total Pages

-----

Any thoughts on this @bmcgrady-ep?

**Comment by bsmartin-ep** [comment]
@srsuddath - looks like `search-selection` is for the resources page.  I'll get that into a custom dimension.

If this looks good on our end, should we this out to production?

**Comment by srsuddath** [comment]
@bsmartin-ep unfortunately we are having to do a rework due to some accessibility issues. Once we have those fixed we will be ready to roll out!

**Comment by jonwehausen** [comment]
Hi @srsuddath, 

Please see the data layer test scenarios below for correction, thanks in advance. 

Test Case | Data Layer Screenshot | Notes
------------ | ------------- | -------------
_Searched "testing" and clicked on the top result_ | ![image](https://user-images.githubusercontent.com/48527022/102498015-6b2e8a80-4047-11eb-995a-719bdd0b7ebc.png) | `null` populates when either of the top recommendations are clicked, and `1` populates when the first result below the top recommendations is clicked. 

**Is it possible for us to always capture the top result as 1 regardless of whether we're given a top recommendation? Thanks.**

**Comment by jonwehausen** [comment]
Hopped on a quick zoom call with @srsuddath to discuss the remaining data layer changes prior to Monday phased-roll out. 

List of DL updates discussed: 
1. For `onsite-search-results-click` -- always start with 1 from the top result and proceed sequentially...if it's a top recommendation, also add additional variable for `search-results-top-recommendation: true|false`
2. For `onsite-search-results-click` -- Add additional variable to populate with the number the user is currently on when clicking a search result...variable name = `search-results-pagination-current-page`
3. For `onsite-search-results-click` -- correct bug for `search-results-total-pages` to populate with the total number of pagination pages possible
4. For `onsite-search-results-click` -- check feature flag for now for type ahead enablement

**Comment by srsuddath** [comment]
Waiting on a final PR which should be merged this afternoon, then @jonwehausen can review and we can finally close this ticket!

**Comment by jonwehausen** [comment]
## QA Testing

### Bug Correction 
Description of Use Case | Data Layer Screenshot | GTM Tag Screenshot | Test Status
------------ | ------------- | ------------- | -------------
Total search result pages | ![image](https://user-images.githubusercontent.com/48527022/102814557-d609f980-4398-11eb-8912-50e79d9c54c9.png) | ![image](https://user-images.githubusercontent.com/48527022/102814610-e5894280-4398-11eb-9fac-c3ea32a97c59.png) | ✔️ **PASS** <br><br>_when there's 999+ results..., 100 will populate for the search result pages_ and 999 will populate for total results


Description of Use Case | Data Layer Screenshot | GTM Tag Screenshot | Test Status
------------ | ------------- | ------------- | -------------
Click on a search result from 3rd page results 3rd position | ![image](https://user-images.githubusercontent.com/48527022/102813875-9e4e8200-4397-11eb-9683-9b1f01d4030a.png) | ![image](https://user-images.githubusercontent.com/48527022/102813907-ab6b7100-4397-11eb-83be-0d2b48fb72b4.png) | ✔️ **PASS** <br><br> **(not set) will populate when `'search-result-chosen-page-url'` becomes available in the next merge
Click on the first search result in top recommendations | ![image](https://user-images.githubusercontent.com/48527022/102814074-fc7b6500-4397-11eb-8222-4ac25088708d.png) | ![image](https://user-images.githubusercontent.com/48527022/102814104-08672700-4398-11eb-9003-df4c3c8c2256.png) ![image](https://user-images.githubusercontent.com/48527022/102814130-0dc47180-4398-11eb-8cfd-b9b8dff3d0b7.png) | ✔️ **PASS** _see custom metrics list below_
Click on a search result just below the top results | ![image](https://user-images.githubusercontent.com/48527022/102814347-74e22600-4398-11eb-95b7-36f09c89fbc5.png) | ![image](https://user-images.githubusercontent.com/48527022/102814358-7f042480-4398-11eb-8328-0ee9727948f6.png) | ✔️ **PASS** <br><br> _top result custom metric did not populate as expected, search result position did


### FYSA Custom Metric Reference
![image](https://user-images.githubusercontent.com/48527022/102814279-4fedb300-4398-11eb-98f8-5a0bc597984a.png)

**Comment by jonwehausen** [comment]
### Edge Case 

@srsuddath Here is another edge case I found we will want to correct: 

_When the user starts typing and is given their options, if the user uses only the keyword and arrow keys down to navigate to their search option and selects one with enter, multiple `view_search_results` events will fire_ 

Let me know if you need me to demo / reproduce this for you

Thanks again for all of your work here @srsuddath !

**Comment by johnhashva** [comment]
@jonwehausen thanks for the deep diving here to make sure we have everything in place and working.   Can you point me/us to how to see Type Ahead clicks in GA?  I know we don't have a dashboard quite yet -- but just looking for raw data collection, if available.

**Comment by jonwehausen** [comment]
Hi @johnhashva ,

Please see the below table for several custom reports for your exploratory analysis. 

Some of this may look messy in GA, it gives us the opportunity to structure this data more effectively in BigQuery and increasing our analysis capabilities. 

You will notice some rows of the event action have `|` instead of `~` this was because we changed the delimiter to `~` in order to make things easier when parsing out in BigQuery. Events will be delimited with `~` moving forward. 

These are each custom reports templates -- when prompted you will want to add to the **All VA.gov Website Data [Production]** view.

**Within each of the "Event Actions" the text after the very last `~` represents either the characters the user typed before selecting a type ahead option OR (_when not selecting a type ahead option_) the query the user ultimately used to perform their search** 

### Searches
Report Link | Report Description
------------ | -------------
[Report Link](https://analytics.google.com/analytics/web/template?uid=tJAqBrzfS1aUQYyZQwbbrw) | Overall volume of with type ahead functionality enabled (represents the 5%)
[Report Link](https://analytics.google.com/analytics/web/template?uid=oAVC8ZnfQWKCRMndUvmDxQ) | Type Ahead Searches by Page: Breaks down from what page type ahead searches have occurred
[Report Link](https://analytics.google.com/analytics/web/template?uid=ZkIm3IvPSa--O4jl4Xrb6Q) | Filtered at the report level for where type ahead is enabled, and the user was given type ahead options (_Type Ahead Options List_) <br><br> `(not set)` indicates a user searched organically and ignored the options they were presented

## Search result clicks
- You will start to see events with both the search result label and search result URL reflected in the event action

Report Link | Report Description
------------ | -------------
[Report Link](https://analytics.google.com/analytics/web/template?uid=YNQ_mMvzQu6ze4uGpqFfdQ) | **Search result clicks** -- _we will need to setup a Google Analytics sequence segment and/or write the query in BQ to segment those users who definitely used type ahead before selecting a search result, the type ahead enabled you see in the event action is currently only looking at the feature flag, not an indicator if it was used_

This is only scratching the surface of the data we have, as there are numerous other reports we could construct that will look better in BQ/DOMO where will be able to leverage the hit-level data and better visualization capabilities...the session aggregation the GA interface performs can be not-so-ideal for this level of customization. 

Thinking forward toward your teams KPIs for this product, if you have some requests of specific reports you know you will be interested for now and rolling into your KPIs, we can get started on standing those up in DOMO. 

Thanks @johnhashva

**Comment by johnhashva** [comment]
Thanks @jonwehausen for this quick work.  Will have time tomorrow to take a deeper look with the team.   Great to have an initial view of usage.

**Comment by denisecoveyduc** [comment]
Removed the S+D Milestone since this isn't actually an issue for the Search & Discovery team but more for analytics.

**Comment by johnhashva** [comment]
@jonwehausen (with @denisecoveyduc FYSA) I finally had time to look at the GA data we are collecting for Type Ahead POC.  Believe I have a good grasp of how to interpret what I'm seeing -- but want to confirm a few things (see attached for PDF of my sum-up).   And from there, we can lock down the DOMO dashboard (BQ) specifications for this.   Can I set time with you this week or early next?  

[VA.gov Type Ahead POC Analytics Update_01052020.pdf](https://github.com/department-of-veterans-affairs/va.gov-team/files/5771880/VA.gov.Type.Ahead.POC.Analytics.Update_01052020.pdf)

**Comment by jonwehausen** [comment]
Hi @johnhashva , 

Thanks a lot here, will give this a review and get back with you. 

And yes certainly, can we shoot for Monday/Tuesday of next week? Thanks in advance

**Comment by jonwehausen** [comment]
Hi @johnhashva , 

We have been configuring queries & views within BigQuery to compare our analysis with the reports here. 

**Date Range: 12/23/2020 - 01/04/2021**
1) _Overall Volume_: Whether it was used or not, type ahead feature flag was enabled for **535,438 unique site visits**, an average of **44,619 site visits per day**
2) _Type Ahead Searches by Page_: We were able to tighten this up a bit more in BigQuery to isolate those searches to only where an options were presented (whether they were chosen or not) at the time the search occurred. Interestingly enough, the 4th highest type ahead searches are coming from the `/search` page, meaning the user elected to use the header search (with type ahead) over the search bar on screen (no type ahead)
![image](https://user-images.githubusercontent.com/48527022/103777139-6fbee100-4ffe-11eb-88f7-0f805b989849.png)
3) _Type Ahead Usage_: Of the total **2502 searches over** this duration where the user was presented a type ahead option in the search dropdown, the option was ignored 31.6% of the time 
4) _Top Type Ahead Options Selected_: The top keywords that were chosen from type ahead options screenshot is below. 
![image](https://user-images.githubusercontent.com/48527022/103778266-fb853d00-4fff-11eb-81dc-bc688ffdba9e.png)

Hope this helps solidify your interpretations around initial usage. From here we can start to dig deeper into how type ahead is influencing search result selections. 

If there are additional pieces of reporting you are interested in prior to our meeting, let us know and we can try to have that prepared for our call.

**Comment by ncksllvn** [comment]
So cool!!!

**Comment by johnhashva** [comment]
Thanks for this refinement @jonwehausen  Def some interesting initial impressions.  I will update PPT to reflect.  And I've set us up for a quick sync on this early next week.

**Comment by jonwehausen** [comment]
@srsuddath As far as implementation is concerned, the final two pieces I have before we can close this issue out:
- Correct `'search-typeahead-enabled'` not populating on `/search/` page searches
- A new variable to add to `onsite-search-results-click` event push -- `'search-result-type'`. This comes as an addition to various types of search results that are now presented on `/find-forms`. In our case for type ahead and regular search we will just always push `'title'` since there's only text links as results there

No rush on these, but following those updates we will be good to close 👍

**Comment by johnhashva** [comment]
@jonwehausen quick interpretation question about this one:  
_Type Ahead Usage: Of the total 2502 searches over this duration where the user was presented a type ahead option in the search dropdown, the option was ignored 31.6% of the time_

Does this mean: User activated the Type Ahead feature by typing in the Search box in the header, but ignored it (did not click on any of the suggested terms)?

**Comment by jonwehausen** [comment]
Hi @johnhashva , 

Yes that is correct, the user had type ahead options on page when they performed their search.

**Comment by jonwehausen** [comment]
Hey @srsuddath , 

Could you provide an update on the status of the remaining updates outlined here? https://github.com/department-of-veterans-affairs/va.gov-team/issues/13494#issuecomment-755349423 

Thanks again for all of your work here.

**Comment by denisecoveyduc** [comment]
Hi @jonwehausen Thanks for reaching out. @srsuddath has been working on the click tracking integration for us so these updates had to be planned for Sprint 39 (#18174 for tracking purposes). As a result, it is in our plan and should be completed by end of sprint (Jan 26th) at the latest.

**Comment by jonwehausen** [comment]
Thanks @denisecoveyduc , very much appreciated!

**Comment by jonwehausen** [comment]
### Documentation of Updates


Description of Update | Data Layer | Test Status
| -- | -- | -- |
 `phased-roll-out-enabled` event firing only once per session now to cut down on hit volume for understanding users who have type ahead enabled 👍 | ![image](https://user-images.githubusercontent.com/48527022/106044133-536a0d80-60ad-11eb-9dac-d37b6a976d6f.png) | ✔️ **PASS**
Number of search results (`search-results-total-count` & `search-results-total-pages` on secondary searches from the `/search` page are now accurate | ![image](https://user-images.githubusercontent.com/48527022/106043587-a394a000-60ac-11eb-849d-c1de5f2a7867.png) | ✔️ **PASS**
`'search-result-type'` added to `'onsite-search-results-click'` for data consistency purposes | ![image](https://user-images.githubusercontent.com/48527022/106043587-a394a000-60ac-11eb-849d-c1de5f2a7867.png) | ✔️ **PASS**
`'search-typeahead-enabled'` now populate on `/search` page | ![image](https://user-images.githubusercontent.com/48527022/106044304-8f9d6e00-60ad-11eb-8ec2-05671d27fe7c.png) | ✔️ **PASS**
